frame/collective/src/lib.rs

```diff
@@ -182,72 +306,200 @@ decl_module! {
+		///
+		/// # <weight>
+		/// ## Weight
+		/// - `O(MP + N)` where:
+		///   - `M` old-members-count (code- and governance-bounded)
+		///   - `N` new-members-count (code- and governance-bounded)
+		///   - `P` proposals-count (code-bounded)
+		/// - DB:
+		///   - 1 storage mutation (codec `O(M)` read, `O(N)` write) for reading and writing the members
+		///   - 1 storage read (codec `O(P)`) for reading the proposals
+		///   - `P` storage mutations (codec `O(M)`) for updating the votes for each proposal
+		///   - 1 storage write (codec `O(1)`) for deleting the old `prime` and setting the new one
+		/// - Benchmark: 0 + M * 20.47 + N * 0.109 + P * 26.29 µs (min squares analysis)
+		/// # </weight>
+		#[weight = (
+			weight_for::set_members(
+				T::DbWeight::get(),
+				T::MaxMembers::get(), // M
+				new_members.len() as Weight, // N
+				T::MaxProposals::get(), // P
+			),
+			DispatchClass::Operational
+		)]
+		fn set_members(
+			origin, new_members: Vec<T::AccountId>, prime: Option<T::AccountId>
+		) -> DispatchResultWithPostInfo {
 			ensure_root(origin)?;
+			ensure!(new_members.len() <= T::MaxMembers::get() as usize, Error::<T, I>::TooManyMembers);
+
+			let old = Members::<T, I>::get();
 			let mut new_members = new_members;
 			new_members.sort();
-			let old = Members::<T, I>::get();
 			<Self as ChangeMembers<T::AccountId>>::set_members_sorted(&new_members[..], &old);
 			Prime::<T, I>::set(prime);
+
+			Ok(Some(weight_for::set_members(
+				T::DbWeight::get(),
+				old.len() as Weight, // M
+				new_members.len() as Weight, // N
+				T::MaxProposals::get(), // P
+			)).into())
 		}
 
 		/// Dispatch a proposal from a member using the `Member` origin.
 		///
 		/// Origin must be a member of the collective.
-		#[weight = (100_000_000, DispatchClass::Operational)]
-		fn execute(origin, proposal: Box<<T as Trait<I>>::Proposal>) {
+		///
+		/// # <weight>
+		/// ## Weight
+		/// - `O(M + P)` where `M` members-count (code-bounded) and `P` complexity of dispatching `proposal`
+		/// - DB: 1 read (codec `O(M)`) + DB access of `proposal`
+		/// - 1 event
+		/// - Benchmark: 22.62 + M * 0.115 + B * 0.003 µs (min squares analysis)
+		/// # </weight>
+		#[weight = (
+			weight_for::execute(T::DbWeight::get(), T::MaxMembers::get(), proposal.get_dispatch_info().weight),
+			DispatchClass::Operational
+		)]
+		fn execute(origin, proposal: Box<<T as Trait<I>>::Proposal>) -> DispatchResultWithPostInfo {
 			let who = ensure_signed(origin)?;
-			ensure!(Self::is_member(&who), Error::<T, I>::NotMember);
+			let members = Self::members();
+			ensure!(members.contains(&who), Error::<T, I>::NotMember);
 
 			let proposal_hash = T::Hashing::hash_of(&proposal);
-			let ok = proposal.dispatch(RawOrigin::Member(who).into()).is_ok();
-			Self::deposit_event(RawEvent::MemberExecuted(proposal_hash, ok));
+			let result = proposal.dispatch(RawOrigin::Member(who).into());
+			Self::deposit_event(RawEvent::MemberExecuted(proposal_hash, result.is_ok()));
```

Comment:
I think we can change `MemberExecuted` and `Executed` to return the dispatch result rather than just a bool.

We made a similar change here: https://github.com/paritytech/substrate/pull/5804

Reply:
Might be worth confirming